File: Model/StayThematic.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Model
{
    [Table("stay_thematic")]
    public class StayThematic
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Column("id")]
        public int? Id { set; get; }

        [Column("stay_id")]
        public int? StayId { set; get; }
        public Stay Stay { set; get; }

        [Column("thematic_id")]
        public int? ThematicId { set; get; }
        public Thematic Thematic { set; get; }
    }
}

File: WebAPP/ClientApp/src/dto/stayThematic.ts
import { Stay } from "./stay";
import { Thematic } from "./thematic";

export class StayThematic {
    public id: number;
    public stayId: number;
    public stay: Stay;
    public thematicId: number;
    public thematic: Thematic

    constructor(id?: number, stayId?: number, stay?: Stay, thematicId?: number, thematic?: Thematic)

    constructor(id: number, stayId: number, stay: Stay, thematicId: number, thematic: Thematic) {
        this.id = id;
        this.stayId = stayId;
        this.stay = stay;
        this.thematicId = thematicId;
        this.thematic = thematic;
    }
}
